tetra/components/base.py

```diff
@@ -703,6 +718,11 @@ class FormComponent(Component, metaclass=FormComponentMetaClass):
 
     _form: Form = None
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.request.method == 'POST':
```

Comment:
This shouldn't be checked here again. When coming from  component_method(), only POSTs are accepted anyway. And if coming from as_tag() (when GET called) it is false.
but submit should not be called from __init__()